src/common/cli_wrapper.h

```diff
@@ -36,15 +44,30 @@ class CLIFormatter : public CLI::Formatter {
   size_t screenWidth_{0};
 };
 
+/**
+ * @brief The class used to define and parse command-line arguments.
+ *
+ * It is a wrapper around https://github.com/CLIUtils/CLI11 that stores defined
+ * command-line arguments in a YAML object.
+ *
+ * Usage outline: first call add() methods to create all the options; then call
+ * parse(argv, argc) to parse command line and get defined options and their
+ * values in a YAML object. The object can be also obtained later by calling
+ * getConfig().
+ *
+ * Options are organized in option groups. Each option group has a header that
+ * preceeds all options in the group. The header for the default option group
+ * can be set from the class constructor.
+ */
 class CLIWrapper {
 private:
-  // Stores option variables
+  // Map with option names and variables
```

Comment:
Don't want to be nit-picky, but an even clearer way would be to say what gets mapped to what. Here is how I woudl write it:

    std::map<std::string, Ptr<any_type>> vars_;   // [option name] -> option value

and maybe rename `vars_` to be more specific, e.g. `allVars_` or whatever it contains.

Reply:
Done